lisa/features/security_profile.py

```
@@ -32,7 +33,11 @@ class SecurityProfile(Feature):
     @classmethod
     def on_before_deployment(cls, *args: Any, **kwargs: Any) -> None:
         settings = cast(SecurityProfileSettings, kwargs.get("settings"))
-        if settings.secure_boot_enabled:
+        runbook: Any = cast(Any, kwargs.get("arm_parameters")).nodes[0]
+        if runbook.security_profile != constants.SECURITY_PROFILE_NONE:
+            settings.security_profile = runbook.security_profile
+            cls._enable_secure_boot(*args, **kwargs)
+        elif settings.security_profile != constants.SECURITY_PROFILE_NONE:
```

Please throw an error if there is no match

---

Closing this comment as it is outdated